internal/api/health.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2023 - MinIO, Inc. All rights reserved.
+// Use of this source code is governed by the AGPLv3
+// license that can be found in the LICENSE file.
+
+package api
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/minio/kes/internal/auth"
+)
+
+func edgeLive(config *EdgeRouterConfig) API {
+	var (
+		Method  = http.MethodGet
+		APIPath = "/v1/alive"
+		MaxBody int64
+		Timeout = 15 * time.Second
+		Verify  = true
+	)
+	if c, ok := config.APIConfig[APIPath]; ok {
+		if c.Timeout > 0 {
+			Timeout = c.Timeout
+		}
+		Verify = !c.InsecureSkipAuth
+	}
+
+	var handler http.HandlerFunc = func(w http.ResponseWriter, r *http.Request) {
+		if err := auth.VerifyRequest(r, config.Policies, config.Identities); Verify && err != nil {
```

Comment:
Does this mean even if the policies and identities fail. The alive and ready will always return 200-OK ?

Reply:
The behavior is:
If auth is enabled (default) we check whether the client has permissions to check liveness resp. readiness.
If auth is disabled for this particular API via: https://github.com/minio/kes/blob/master/server-config.yaml#L76
any client can query the liveness resp. readiness API.

Currently, there is no concept of non-initialized IAM - like in MinIO. Either the KES config file is valid and the KES server starts with initialized IAM or it refuses to start.

Reply:
Yes but liveness and readiness must also ensure that stateless kes can access it's master key. 

Its not just about Kes is up or not - it is about if Kes can reach it's master key. So we need to add that.

Reply:
Okay, in that case we can query the KMS state in the readiness endpoint. While KES could still be able to process requests even if its backend KMS is down - e.g. offline cache - it seems reasonable to fail since at least some requests - e.g. create key - cannot be processed.